docs/modules/serialization/pages/compact-serialization.adoc

```diff
@@ -35,368 +35,1501 @@ a class to implement a particular interface. Serializers might be implemented an
+that their read and write methods are compatible, and they have the same type name.
 
-    public String getName() {
-        return name;
-    }
-}
-----
+NOTE: If you evolve your class in the later versions of your application, by adding
+or removing fields, you should continue using the same type name for that class.
 
-If you don't perform any kind of configuration changes and use the instances of the class
-directly, no exceptions are thrown. Hazelcast will generate a schema out of the
-`Employee` class the first time you try to serialize an object, cache it, and reuse it
-for the subsequent serializations and deserializations.
+To register an explicit serializer for a certain class, add the following configuration:
 
-The same holds true for the Java records. Hazelcast supports serializing and deserializing
-Java records, without an extra configuration as well.
+**Programmatic Configuration:**
 
-Assuming the `Employee` class above were a Java record:
+[tabs]
+====
 
+Java - Member::
++
+--
 [source,java]
 ----
-public record Employee(long id, String name) {
-}
+Config config = new Config();
+config.getSerializationConfig()
+        .getCompactSerializationConfig()
+        .addSerializer(new FooSerializer());
 ----
+--
 
-The following code would work for both of them.
-
+Java - Client::
++
+--
 [source,java]
 ----
-HazelcastInstance client = HazelcastClient.newHazelcastClient();
-IMap<Long, Employee> map = client.getMap("employees");
-Employee employee = new Employee(1, "John Doe");
-map.set(1L, employee);
-Employee employeeFromMap = map.get(1L);
+ClientConfig clientConfig = new ClientConfig();
+clientConfig.getSerializationConfig()
+        .getCompactSerializationConfig()
+        .addSerializer(new FooSerializer());
 ----
+--
 
-== Implementing CompactSerializer
-
-Another way to use compact serialization is to implement the `CompactSerializer` interface for a class
-and register it in the configuration.
-
-For example, assume that you have the same `Employee` class. Then, a Compact serializer can be implemented as such.
-
-[source,java]
+.NET::
++
+--
+[source,cs]
 ----
-public class EmployeeSerializer implements CompactSerializer<Employee> {
-    @Override
-    public Employee read(CompactReader reader) {
-        long id = reader.readInt64("id");
-        String name = reader.readString("name");
-        return new Employee(id, name);
-    }
-
-    @Override
-    public void write(CompactWriter writer, Employee employee) {
-        writer.writeInt64("id", employee.getId());
-        writer.writeString("name", employee.getName());
-    }
-
-    @Override
-    public Class<Employee> getCompactClass() {
-        return Employee.class;
-    }
 
-    @Override
-    public String getTypeName() {
-        return "employee";
-    }
-}
 ----
+--
 
-The last step is to register the serializer in the member or client configuration.
+Node.js::
++
+--
+[source,javascript]
+----
+Client.newHazelcastClient(
```

Comment:
@srknzl @harunalpak I assume this is a Javascript code, not TypeScript. Is it OK for the code samples? Same for the other Node.js code snippets

Reply:
IMO, using javascript in code samples is better. Using our client with typescript is just an option. We use javascript in our code samples as well. Typescript users can easily convert these examples to TS.